plugins/modules/mysql_role.py

```diff
@@ -0,0 +1,784 @@
+            if user not in users:
+                if check_mode:
+                    return True
+
+                if not self.is_mariadb:
+                    self.cursor.execute('REVOKE %s@%s FROM %s@%s', (self.name, self.host, user[0], user[1]))
+                else:
+                    self.cursor.execute('REVOKE %s FROM %s@%s', (self.name, user[0], user[1]))
+                changed = True
+
+        return changed
+
+    def remove_members(self, users, check_mode=False):
+        """Remove members from a role.
+
+        Args:
+            users (list): Role members.
+            check_mode (bool): If True, just checks and does nothing.
+
+        Returns:
+            bool: True if the state has changed, False if has not.
+        """
+        if not users:
+            return False
+
+        changed = False
+        for user in users:
+            if user in self.members:
+                if check_mode:
+                    return True
+
+                if not self.is_mariadb:
+                    self.cursor.execute('REVOKE %s@%s FROM %s@%s', (self.name, self.host, user[0], user[1]))
+                else:
+                    self.cursor.execute('REVOKE %s FROM %s@%s', (self.name, user[0], user[1]))
+                changed = True
+
+        return changed
+
+    def update(self, users, privs, check_mode=False,
+               append_privs=False, append_members=False,
+               detach_members=False, admin=False,
+               set_default_role_all=True):
+        """Update a role.
+
+        Update a role if needed.
+
+        Todo: Implement changing of role's admin when ALTER ROLE statement
+            to do that gets supported.
+
+        Args:
+            users (list): Role members.
+            privs (str): String containing privileges.
+            check_mode (bool): If True, just checks and does nothing.
+            append_privs (bool): If True, adds new privileges passed through privs
+                not touching current privileges.
+            append_members (bool): If True, adds new members passed through users
+                not touching current members.
+            detach_members (bool): If True, removes members passed through users from a role.
+            admin (list): Role's admin. Contains (username, hostname).
+            set_default_role_all (bool): If True, runs SET DEFAULT ROLE ALL TO each member.
+
+        Returns:
+            bool: True if the state has changed, False if has not.
+        """
+        changed = False
+
+        if users:
+            if detach_members:
+                changed = self.remove_members(users, check_mode=check_mode)
+
+            else:
+                changed = self.add_members(users, check_mode=check_mode,
+                                           append_members=append_members,
+                                           set_default_role_all=set_default_role_all)
+
+        if privs:
+            changed, msg = user_mod(self.cursor, self.name, self.host,
```

Comment:
It looks like when both `users` and `privs` are set, you override the value set by the previous block for `changed`.
Ex. when you add a member but there is no change in the role privileges, you would return `False` instead of `True`.

Reply:
very good spot, thanks! fixed